pages/nft/url/index.vue

```diff
@@ -207,57 +219,82 @@ export default class FetchIndex extends Vue {
     this.arweaveId = ''
     this.arweaveFeeTxHash = ''
     this.iscnId = ''
+    this.state = State.INIT
   }
 
   async onSubmit() {
-    if (this.ownerWallet && this.address !== this.ownerWallet) {
-      this.errorMessage = 'PLEASE_USE_OWNER_WALLET_TO_SIGN'
-      return
-    }
-    if (this.isUrlIscnId) {
-      this.iscnId = this.url;
-      this.$router.push(
-        this.localeLocation({
-          name: 'nft-iscn-iscnId',
-          params: this.iscnParams,
-        })!,
-      )
-      return;
-    }
-
-    this.balance = await getAccountBalance(this.address) as string
-    if (this.balance === '0') {
-      this.toggleSnackbar('INSUFFICIENT_BALANCE')
-      return
-    }
-
     try {
-      this.errorMessage = ''
-      if (!this.url) {
-        this.errorMessage = this.$t(
-          'HomePage.search.errormessage.empty',
-        ) as string
-        return
-      }
-      this.isLoading = true
-      await this.crawlUrlData()
-      if (this.crawledData?.body) {
-          const arweaveFeeInfo = await this.estimateArweaveFee()
-          if (!this.arweaveId) {
-            if (!this.arweaveFeeTxHash) { await this.sendArweaveFeeTx(arweaveFeeInfo) }
-            await this.submitToArweave()
-          }
+      while (this.state !== State.DONE) {
+        // eslint-disable-next-line no-await-in-loop
+        await this.doAction()
       }
-      await this.submitToISCN()
     } catch (err) {
-      // eslint-disable-next-line no-console
-      console.error(err)
       this.setError(err)
     } finally {
       this.isLoading = false
     }
   }
 
+  onSkip() {
+    this.state = State.TO_REGISTER
+    this.onSubmit()
+  }
+
+  async doAction() {
+    this.isLoading = true
+    switch (this.state) {
+      case State.INIT:
+        if (this.ownerWallet && this.address !== this.ownerWallet) {
+          this.errorMessage = 'PLEASE_USE_OWNER_WALLET_TO_SIGN'
+          break
+        }
+        if (this.isUrlIscnId) {
+          this.iscnId = this.url;
+          this.$router.push(
+            this.localeLocation({
+              name: 'nft-iscn-iscnId',
+              params: this.iscnParams,
+            })!,
+          )
+          break
+        }
+
+        this.balance = await getAccountBalance(this.address) as string
+        if (this.balance === '0') {
+          this.toggleSnackbar('INSUFFICIENT_BALANCE')
+          break
+        }
+
+        this.errorMessage = ''
+        if (!this.url) {
+          this.errorMessage = this.$t('HomePage.search.errormessage.empty') as string
+          break
+        }
+        this.state = State.TO_CRAWL
+        break
+      case State.TO_CRAWL:
+        await this.crawlUrlData()
+        this.state = State.TO_ESTIMATE_FEE
+        break
```

Comment:
If we don't `break` after each sucessful `State` and only break on error, we don't need to use a while loop, just call this function will do

Reply:
Yes, that's smarter!